<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>637</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RitM in the Middle</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res.qrc">
    <normaloff>:/images/logo.png</normaloff>:/images/logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="ProcTableWidget" name="tableWidget_field">
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="acceptDrops">
       <bool>true</bool>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="tabKeyNavigation">
       <bool>false</bool>
      </property>
      <property name="dragEnabled">
       <bool>true</bool>
      </property>
      <property name="dragDropMode">
       <enum>QAbstractItemView::DragDrop</enum>
      </property>
      <property name="defaultDropAction">
       <enum>Qt::MoveAction</enum>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectItems</enum>
      </property>
      <property name="iconSize">
       <size>
        <width>60</width>
        <height>60</height>
       </size>
      </property>
      <property name="verticalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
      <property name="horizontalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
      <property name="showGrid">
       <bool>true</bool>
      </property>
      <property name="gridStyle">
       <enum>Qt::DashLine</enum>
      </property>
      <property name="rowCount">
       <number>32</number>
      </property>
      <property name="columnCount">
       <number>32</number>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>64</number>
      </attribute>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>64</number>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderDefaultSectionSize">
       <number>64</number>
      </attribute>
      <attribute name="verticalHeaderMinimumSectionSize">
       <number>64</number>
      </attribute>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <row/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
      <column/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1002</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="action_new"/>
    <addaction name="action_open"/>
    <addaction name="action_save"/>
    <addaction name="action_saveas"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="action_quit"/>
   </widget>
   <widget class="QMenu" name="menu_edit">
    <property name="title">
     <string>Правка</string>
    </property>
    <addaction name="action_cut"/>
    <addaction name="action_copy"/>
    <addaction name="action_paste"/>
    <addaction name="separator"/>
    <addaction name="action_delete"/>
    <addaction name="action_selectall"/>
   </widget>
   <widget class="QMenu" name="menu_widgets">
    <property name="title">
     <string>Окна</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_info">
    <property name="title">
     <string>Справка</string>
    </property>
    <addaction name="action_help"/>
    <addaction name="action_about"/>
   </widget>
   <widget class="QMenu" name="menu_view">
    <property name="title">
     <string>Вид</string>
    </property>
    <addaction name="action_zoomin"/>
    <addaction name="action_zoomout"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Процесс</string>
    </property>
    <addaction name="action_check"/>
    <addaction name="action_start"/>
    <addaction name="action_stop"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_view"/>
   <addaction name="menu_edit"/>
   <addaction name="menu"/>
   <addaction name="menu_widgets"/>
   <addaction name="menu_info"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>Панель инструментов</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonIconOnly</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_new"/>
   <addaction name="action_open"/>
   <addaction name="action_save"/>
   <addaction name="separator"/>
   <addaction name="action_zoomin"/>
   <addaction name="action_zoomout"/>
   <addaction name="separator"/>
   <addaction name="action_cut"/>
   <addaction name="action_copy"/>
   <addaction name="action_paste"/>
   <addaction name="separator"/>
   <addaction name="action_check"/>
   <addaction name="action_start"/>
   <addaction name="action_stop"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockWidget_elements">
   <property name="windowTitle">
    <string>Библиотечные обработчики</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="ProcListWidget" name="listWidget_elements">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="iconSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
       <property name="textElideMode">
        <enum>Qt::ElideRight</enum>
       </property>
       <property name="movement">
        <enum>QListView::Free</enum>
       </property>
       <property name="resizeMode">
        <enum>QListView::Adjust</enum>
       </property>
       <property name="gridSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="viewMode">
        <enum>QListView::IconMode</enum>
       </property>
       <property name="uniformItemSizes">
        <bool>true</bool>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
       <property name="selectionRectVisible">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>Base.Adapter</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/adapter.svg</normaloff>:/images/adapter.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Base.Connector</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/connector.svg</normaloff>:/images/connector.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Base.Processor</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/processor.svg</normaloff>:/images/processor.svg</iconset>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Размер иконок:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="horizontalSlider_elements">
         <property name="minimum">
          <number>32</number>
         </property>
         <property name="maximum">
          <number>80</number>
         </property>
         <property name="value">
          <number>64</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBothSides</enum>
         </property>
         <property name="tickInterval">
          <number>8</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_pipes">
   <property name="windowTitle">
    <string>Соединители обработчиков</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="ProcListWidget" name="listWidget_pipes">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="iconSize">
        <size>
         <width>64</width>
         <height>64</height>
        </size>
       </property>
       <property name="textElideMode">
        <enum>Qt::ElideRight</enum>
       </property>
       <property name="movement">
        <enum>QListView::Free</enum>
       </property>
       <property name="resizeMode">
        <enum>QListView::Adjust</enum>
       </property>
       <property name="gridSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="viewMode">
        <enum>QListView::IconMode</enum>
       </property>
       <property name="uniformItemSizes">
        <bool>true</bool>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
       <property name="selectionRectVisible">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>Pipe.Left2Right</string>
        </property>
        <property name="toolTip">
         <string>Pipe.Left2Right</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/Left2Right.svg</normaloff>:/images/pipes/Left2Right.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Pipe.Left2Top</string>
        </property>
        <property name="toolTip">
         <string>Pipe.Left2Top</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/Left2Top.svg</normaloff>:/images/pipes/Left2Top.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Pipe.Left2Bottom</string>
        </property>
        <property name="toolTip">
         <string>Pipe.Left2Bottom</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/Left2Bottom.svg</normaloff>:/images/pipes/Left2Bottom.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Pipe.Left2TopBottom</string>
        </property>
        <property name="toolTip">
         <string>Pipe.Left2TopBottom</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/Left2TopBottom.svg</normaloff>:/images/pipes/Left2TopBottom.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Pipe.Top2Right</string>
        </property>
        <property name="toolTip">
         <string>Pipe.Top2Right</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/Top2Right.svg</normaloff>:/images/pipes/Top2Right.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Pipe.Bottom2Right</string>
        </property>
        <property name="toolTip">
         <string>Pipe.Bottom2Right</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/Bottom2Right.svg</normaloff>:/images/pipes/Bottom2Right.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Pipe.TopBottom2Right</string>
        </property>
        <property name="toolTip">
         <string>Pipe.TopBottom2Right</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/TopBottom2Right.svg</normaloff>:/images/pipes/TopBottom2Right.svg</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Pipe.Top2Bottom</string>
        </property>
        <property name="toolTip">
         <string>Pipe.Top2Bottom</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/images/pipes/Top2Bottom.svg</normaloff>:/images/pipes/Top2Bottom.svg</iconset>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Размер иконок:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="horizontalSlider_pipes">
         <property name="minimum">
          <number>32</number>
         </property>
         <property name="maximum">
          <number>80</number>
         </property>
         <property name="value">
          <number>64</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBothSides</enum>
         </property>
         <property name="tickInterval">
          <number>8</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_undo">
   <property name="windowTitle">
    <string>Список изменений</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_2"/>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Файл журнала</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QTextBrowser" name="textBrowser_log">
       <property name="lineWrapMode">
        <enum>QTextEdit::NoWrap</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_open">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/fileopen.png</normaloff>:/images/fileopen.png</iconset>
   </property>
   <property name="text">
    <string>Открыть ...</string>
   </property>
   <property name="toolTip">
    <string>Открыть</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_save">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/filesave.png</normaloff>:/images/filesave.png</iconset>
   </property>
   <property name="text">
    <string>Сохранить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_saveas">
   <property name="text">
    <string>Сохранить как ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_quit">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/quit.png</normaloff>:/images/quit.png</iconset>
   </property>
   <property name="text">
    <string>Выход</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="action_about">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/about.png</normaloff>:/images/about.png</iconset>
   </property>
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
  <action name="action_help">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/help.png</normaloff>:/images/help.png</iconset>
   </property>
   <property name="text">
    <string>Помощь</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="action_cut">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/editcut.png</normaloff>:/images/editcut.png</iconset>
   </property>
   <property name="text">
    <string>Вырезать</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="action_paste">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/editpaste.png</normaloff>:/images/editpaste.png</iconset>
   </property>
   <property name="text">
    <string>Вставить</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="action_copy">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/editcopy.png</normaloff>:/images/editcopy.png</iconset>
   </property>
   <property name="text">
    <string>Копировать</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="action_zoomin">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/zoomin.png</normaloff>:/images/zoomin.png</iconset>
   </property>
   <property name="text">
    <string>Увеличить масштаб</string>
   </property>
   <property name="shortcut">
    <string>Ctrl++</string>
   </property>
  </action>
  <action name="action_zoomout">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/zoomout.png</normaloff>:/images/zoomout.png</iconset>
   </property>
   <property name="text">
    <string>Уменьшить масштаб</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="action_new">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/filenew.png</normaloff>:/images/filenew.png</iconset>
   </property>
   <property name="text">
    <string>Новый тракт</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_delete">
   <property name="text">
    <string>Удалить выделенные</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_selectall">
   <property name="text">
    <string>Выделить все</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="action_check">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/check.png</normaloff>:/images/check.png</iconset>
   </property>
   <property name="text">
    <string>Проверить схему</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="action_start">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/start.png</normaloff>:/images/start.png</iconset>
   </property>
   <property name="text">
    <string>Запустить обработку</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_stop">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/images/stop.png</normaloff>:/images/stop.png</iconset>
   </property>
   <property name="text">
    <string>Остановить обработку</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ProcTableWidget</class>
   <extends>QTableWidget</extends>
   <header>ProcTableWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ProcListWidget</class>
   <extends>QListWidget</extends>
   <header>ProcListWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>horizontalSlider_elements</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_pipes</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>126</x>
     <y>595</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_pipes</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_elements</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>589</y>
    </hint>
    <hint type="destinationlabel">
     <x>195</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
